lib/utils.js

```diff
@@ -50,6 +51,38 @@ timeout_redirect = function(im) {
     return im.config.timeout_redirects.indexOf(im.user.state.name) !== -1;
 };
 
+// REQUEST HELPERS
+
+/**: get_paginated_response(service, url, params)
+Gets all pages of a paginated request.
+
+:param object service: The api for the service being queried.
+:param string url: The url to send the request to.
+:param object params: The query parameters for the request
+*/
+get_paginated_response = function(service, url, params) {
+    // Get the current page and log the call
+    return service.http_api.get(url, {params: params})
+        .then(service.log_service_call('GET', url, null, params))
+        .then(function(response) {
+            var results = response.data.results;
+            if (response.data.next === null) {
+                return {results: results, count: results.length};
+            }
+            var next_url = url_utils.parse(response.data.next, true);
+            // Recursively get next pages
+            return get_paginated_response(
+                    service, response.data.next, next_url.query).
```

Comment:
I think there may be a problem here but not totally sure. If `response.data.next` is something like `http://localhost:8000/api/v1/subscriptions/?cursor=cD0yMDE3LTA4LTA5KzE0JTNBMTQlM0EyMS4xNDAwMDMlMkIwMCUzQTAw` then the call will include the `cursor` param twice. And it gets a bit more confusing when the query already includes parameters.

Maybe it makes sense for this method to not take a `params`, but then the caller has to organise parameters? I don't know if that's great either :/

Reply:
I was actually having a huge problem here because the stuff under the hood _removes_ the query params that are already part of the url. That's why I have to parse them out and pass then in separately.
In the tests you'll see that the fixtures that match these calls don't have the params in the url.